src/app/app.component.ts

```diff
@@ -21,4 +26,37 @@ export class AppComponent {
       return 'off-home';
     }
   }
+
+  ngDoCheck() {
+    if (this.auth.authenticated() && this.currentUserId == null)
```

Comment:
:+1: i should take the time to study what's happening here so i can offer better advice.